### ISSUE TITLE
4.0.1 typescript error TS2322 on subscriptions

### ISSUE THREAD

**Issue by rosskevin** [final-form/final-form#65]
### Are you submitting a **bug report** or a **feature request**?
Bug

### What is the current behavior?
I introduced changes in #61 that produce typescript errors.

### What is the expected behavior?
One example:

```
error TS2322: Type '{ subscription: { errors: true; values: true; }; children: (form: FormSpyRenderProps) => Element; }' is not assignable to type 'IntrinsicAttributes & FormSpyProps & { children?: ReactNode; }'.
  Type '{ subscription: { errors: true; values: true; }; children: (form: FormSpyRenderProps) => Element; }' is not assignable to type 'FormSpyProps'.
    Types of property 'subscription' are incompatible.
      Type '{ errors: true; values: true; }' is not assignable to type 'FormSubscription'.
        Property 'active' is missing in type '{ errors: true; values: true; }'.
```

### What's your environment?
 final-form        ^4.0.1 
 react-final-form  ^3.0.2 

### Other information
I created the problem, I'll PR the fix ASAP.  Working on it now.

**Comment by erikras** [comment]
Thanks for submitting this issue to alert others that it's being worked on. 👍

**Comment by rosskevin** [comment]
@erikras - notice that an indexer is added to both the flow and ts types for `Subscription`.  I don't think that `any` key is allowed here.  Should it be 100% open or only the known keys?

**Comment by erikras** [comment]
Not sure what you mean by `indexer`. Point me to a line. You mean [this](https://github.com/final-form/final-form/blob/master/src/types.js.flow#L2)?

**Comment by rosskevin** [comment]
Yes exactly, that means it allows _any_ field with a boolean value

**Comment by erikras** [comment]
Aaahh.. It should only be known keys. So really the whole `Subscription` type can go away, I guess. It's not providing any value.

**Comment by rosskevin** [comment]
It is now, I'll adjust flow like ts.  Subscription will be common fields.

**Comment by erikras** [comment]
But why not just have `FormSubscription` and `FieldSubscription`? There's nowhere where we need an object that could be either.

**Comment by rosskevin** [comment]
`FinalForm.js` uses the generic `Subscription` and switching to flow `FormSubscription` is causing errors.  Haven't had a chance to look at it yet.  Will do shortly.

**Comment by rosskevin** [comment]
@erikras I bailed out on the flow type changes.  It started down a rabbit hole that I'm not quite prepared to follow.  I fixed the ts types so we need to mirror them to flow and deal with the subsequent fallout in a different issue/PR.

**Comment by erikras** [comment]
> I bailed out on the flow type changes

I'm okay with that. 👍

**Comment by erikras** [comment]
Published fix in [`v4.0.2`](https://github.com/final-form/final-form/releases/tag/v4.0.2).
